src/components/output/custom-els/MultiPanel/index.ts

```diff
@@ -0,0 +1,220 @@
+import './styles.css';
+
+/**
+ * A multi-panel view that the user can add any number of 'panels'.
+ * 'a panel' consists of two elements. Even index element becomes heading,
+ * and odd index element becomes the expandable content.
+ */
+
+function getClosestHeading(el: Element) {
+  const closestEl = el.closest('multi-panel > *');
+  if (closestEl && closestEl.classList.contains('panel-heading')) {
+    return closestEl;
+  }
+  return undefined;
+}
+
+export default class MultiPanel extends HTMLElement {
+
+  constructor() {
+    super();
+
+    // add EventListeners
+    this.addEventListener('click', this._onClick);
+    this.addEventListener('keydown', this._onKeyDown);
+
+    // Watch for children changes.
+    new MutationObserver(() => this._childrenChange())
+      .observe(this, { childList: true });
+  }
+
+  connectedCallback() {
+    this._childrenChange();
+  }
+
+  // Click event handler
+  private _onClick(event: MouseEvent) {
+    const el: Element = event.target as Element;
+    const heading = getClosestHeading(el);
+    if (!heading) return;
+    this._expand(heading);
+  }
+
+  // KeyDown event handler
+  private _onKeyDown(event: KeyboardEvent) {
+    const selectedEl = document.activeElement;
+    const heading = getClosestHeading(selectedEl);
+
+    // if keydown event is not on heading element, ignore
+    if (!heading) return;
+
+    // don’t handle modifier shortcuts used by assistive technology.
+    if (event.altKey) return;
+
+    let newHeading;
+    switch (event.key) {
+      case 'ArrowLeft':
+      case 'ArrowUp':
+        newHeading = this._prevHeading();
+        break;
+
+      case 'ArrowRight':
+      case 'ArrowDown':
+        newHeading = this._nextHeading();
+        break;
+
+      case 'Home':
+        newHeading = this._firstHeading();
+        break;
+
+      case 'End':
+        newHeading = this._lastHeading();
+        break;
+
+      case 'Enter':
+      case ' ':
+      case 'Spacebar':
+        this._expand(heading);
+        break;
+
+      // Any other key press is ignored and passed back to the browser.
+      default:
+        return;
+    }
+
+    event.preventDefault();
+    selectedEl.setAttribute('tabindex', '-1');
```

Comment:
If `newHeading` is undefined, doesn't this mean we lose the ability to focus any heading, or have I misread things?

Reply:
New heading is always set if it made it thus far in the code. On switch statement part, it is wither setting new hading or hitting space bar to open. 

This is wrapped in if statement to make TypeScript happy. yes, it _could_ be undefined but almost certainly not because any keyboard input that does not set new heading is caught in `default` and returned.

Reply:
oh wait, hmmmmm you are right in case of spacebar, it'll take the focus away I guess. moving that line inside if statement.